packages/nimble-components/src/table/specs/table-column-sort-hld.md

```diff
@@ -67,6 +89,31 @@ Each column that is sorted will have an appropriate icon displayed in the header
 
 Each column that is sorted ascending will have `aria-sort="ascending"` set on the header, and each column that is sorted descending will have `aria-sort="descending"` set on the header.
 
+### Interactive Sorting UX
+
+The UX/ interactions for interactive sorting are captured in [#885: UX Design for basic column/row interactions](https://github.com/ni/nimble/issues/885).  
+To summarize:
+
+If the table `sortMode` is `single` or `multiple`:  
+Single-clicking a column header will cycle the column from unsorted, to ascending sort, descending sort, then back to unsorted.
+
+-   Any other columns that were also sorted become unsorted
+-   The clicked column gets a sort index of `0` when sorted, and `undefined` if it's transitioning back to unsorted.
```

Comment:
sort-index is not incrementing integer values from zero, they are arbitrary ordered numbers. some may be negative so setting zero isn't the highest precedence.

Should we have stricter requirements on the sort-indexes like they must be counted integers starting at zero? 

Is it more sane that during interactive operations we do something like rewrite all the sort-indexes? Or maybe we should behave more like the sizing approach and reflect the table evaluated state to currentSortIndex, currentSortDirection, etc?

Did you have thoughts on this already @mollykreis @atmgrifter00 ?

Reply:
Single-clicking a column unsorts all the other columns. My plan was to both set `sortDirection` to `none` and set `sortIndex` to `undefined` on all the columns we were unsorting. So that would leave only one column with a sort index (`0`) in the single-sort case.  
The same applies when multi-sort is enabled, unless you hold down Shift, which is when I was planning to leave the other sorted columns alone (with their existing sort index), just picking `max(existingIndices) + 1` for the clicked column. So that is one case where you could have nonconsecutive/negative sort indices (if they were previously set programatically). So you might end up with `-5`,`-3`,`-2` sort indices under that logic (`-2` would be what was chosen for the clicked column in that example).

So I think the algorithm I proposed in the HLD is valid even if negative sort indices were already present on some columns. But asking if we should rewrite the indices during any interactive sort is a fair question I think.

Reply:
@msmithNI or @mollykreis, is there a reason that interactive sorting wouldn't behave in a similar fashion as interactive column sizing, with respect to the state that it affects? Perhaps the exact implementation details aren't being alluded to here, but I wanted to make sure we're on the same page about whether an interactive sort change would update the `sort-index` attribute on the column (_not_ aligned with the column sizing behavior), or update some internal property like `current-sort-index` (aligned with the column sizing behavior).

Reply:
Updated the HLD to state that
- interactive sorting changes will not update `sortDirection` / `sortIndex`
- added new properties `currentSortDirection` / `currentSortIndex` which will reflect interactive sort index changes

I thought about rewriting/normalizing the sort indices (which would mean that `currentSortIndex` would always be `undefined, or `0` or greater) but didn't really see a reason to do so. Hence, my previous comment about sort indices / the algorithm being used is still valid for this latest HLD update.

Keeping this conversation thread open to check if people agree (or if they do see a reason to normalize the sort indices).

Reply:
As discussed today, updated the HLD to say that `currentSortIndex` will be normalized to `0..(n-1)` for `n` sorted columns (mainly applies in the Shift-click case where the existing sorted columns are kept).

Also removed the `column-sort-change` event since we don't have an immediate need for it, and there were some unanswered questions around it (like should it give the user the normalized sort indices / we might eventually want it to be cancelable / etc).